File: typescript/01-openai-basic.ts
/**
 * Minimal Senthex integration: swap baseURL, add X-Senthex-Key, done.
 * Run: npx tsx 01-openai-basic.ts
 */
import OpenAI from "openai";

const openaiKey = process.env.OPENAI_API_KEY;
const senthexKey = process.env.SENTHEX_API_KEY;
if (!openaiKey || !senthexKey) {
  throw new Error("OPENAI_API_KEY and SENTHEX_API_KEY must both be set");
}

const client = new OpenAI({
  apiKey: openaiKey,
  baseURL: "https://app.senthex.com/v1",
  defaultHeaders: { "X-Senthex-Key": senthexKey },
});

const { data, response } = await client.chat.completions
  .create({
    model: "gpt-4o-mini",
    messages: [{ role: "user", content: "Say hi in one short sentence." }],
  })
  .withResponse();

console.log(`Request-Id:      ${response.headers.get("x-senthex-request-id")}`);
console.log(`Shield-Status:   ${response.headers.get("x-senthex-shield-status")}`);
console.log(`Injection-Score: ${response.headers.get("x-senthex-injection-score")}`);
console.log(`Latency-Ms:      ${response.headers.get("x-senthex-latency-ms")}`);
console.log();
console.log(data.choices[0]?.message.content ?? "");

File: python/01_openai_basic.py
"""Minimal Senthex integration: swap base_url, add X-Senthex-Key, done.

Reads the verdict from response headers without changing how you call OpenAI.
Run: python 01_openai_basic.py
"""

import os

from openai import OpenAI

client = OpenAI(
    api_key=os.environ["OPENAI_API_KEY"],
    base_url="https://app.senthex.com/v1",
    default_headers={"X-Senthex-Key": os.environ["SENTHEX_API_KEY"]},
)

raw = client.chat.completions.with_raw_response.create(
    model="gpt-4o-mini",
    messages=[{"role": "user", "content": "Say hi in one short sentence."}],
)

print(f"Request-Id:      {raw.headers.get('X-Senthex-Request-Id')}")
print(f"Shield-Status:   {raw.headers.get('X-Senthex-Shield-Status')}")
print(f"Injection-Score: {raw.headers.get('X-Senthex-Injection-Score')}")
print(f"Latency-Ms:      {raw.headers.get('X-Senthex-Latency-Ms')}")
print()

response = raw.parse()
print(response.choices[0].message.content)
